### PR TITLE
Technical guidelines: added guidelines about IDs

### PR DESCRIPTION
**Pull request by buskamuza**
## Purpose of this pull request

This pull request (PR) updates guidelines, specifically about IDs.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/graphql/develop/create-graphqls-file.html
- https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html

Updated the [Technical Guidelines](https://devdocs.magento.com/guides/v2.3/graphql/develop/create-graphqls-file.html) to include the following statement: Service contracts SHOULD allow client side generated IDs.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/coding-standards/technical-guidelines.md [hunk 1]
```diff
@@ -574,6 +574,8 @@ You need to read configuration from different sources (like database or filesyst
 
 6.4.3.8. Data objects returned by service contracts SHOULD be fully loaded to ensure consistency.
 
+6.4.3.9. Service contracts SHOULD allow client side generated IDs. A service SHOULD accept an ID instead of generating it. A service SHOULD use `string` as a type for ID, so that it is compatible with common algorithms for client side generated IDs.
```
Comment (akaplya) [inline guides/v2.2/coding-standards/technical-guidelines.md]: This statement is true for GraphQL schema but not necessary for the service contracts.
In code, client-generated Id, or its particular example UUID could have another datatype than string.

**Comment by buskamuza** [comment]
Discussed on Architecture Discussions meeting. No concerns appeared. The PR can be merged if acceptable by devdocs team.

File: guides/v2.3/graphql/develop/create-graphqls-file.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ In a `schema.graphqls` file, the output `Interface` defines top-level attributes
 
 In many cases, the response contains data that was either not available as input, or was transformed in some manner from the input. For example, when you specify a price in an input filter, Magento evaluates it as a Float value. However, `Price` output objects contain a Float value, a currency value, and possibly minimum/maximum values and tax adjustments. You can define a `typeResolver` to point to the Resolver object, which interprets the GraphQL query. If your module contains only attributes that extend another module, then this parameter is optional. Otherwise, it is required. See [Resolvers]({{ page.baseurl }}/graphql/develop/resolvers.html) for more information.
 
+Output types that represent entities, which should be manipulated (updated, removed, etc.) and/or can be cached on the client, MUST have `id` field. Type of the field SHOULD be `ID`.
```
[the remark this answers is not in the document]
Comment (buskamuza) [reply guides/v2.3/graphql/develop/create-graphqls-file.md]: Looks good. Thanks.
Please let me know if there is anything else necessary from my side for this PR?

**Comment by keharper** [comment]
running tests

**Pull request merged**